### PR TITLE
fix(query):  table field data type in `show create table` incorrect

### PR DESCRIPTION
**Pull request by forsaken628**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

```
create table xx (a array(string),b array(string null),c array(string not null));
show create table xx;

-[ RECORD 1 ]-----------------------------------
       Table: xx
Create Table: CREATE TABLE xx (
  a ARRAY(STRING) NULL,
  b ARRAY(STRING) NULL,
  c ARRAY(STRING) NULL
) ENGINE=FUSE
```

## Tests

- [x] Unit Test
- [ ] Logic Test
- [ ] Benchmark Test
- [ ] No Test - _Explain why_

## Type of change

- [x] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/databendlabs/databend/17112)

### CODE REVIEW

**Comment by forsaken628** [comment]
Any more issues like that?

File: tests/sqllogictests/suites/base/05_ddl/05_0003_ddl_alter_table.test [hunk 1]
```diff
@@ -130,7 +130,7 @@ INSERT INTO TABLE `05_0003_at_t4` values('a', 'b', ['c1', 'c2'], ('d1', 'd2'))
 query TT
 SHOW CREATE TABLE `05_0003_at_t4`
 ----
-05_0003_at_t4 CREATE TABLE "05_0003_at_t4" ( a VARCHAR NOT NULL, b VARCHAR NULL, c ARRAY(STRING) NULL, d TUPLE(1 STRING, 2 STRING) NULL ) ENGINE=FUSE
+05_0003_at_t4 CREATE TABLE "05_0003_at_t4" (   a VARCHAR NOT NULL,   b VARCHAR NULL,   c ARRAY(VARCHAR NULL) NULL,   d TUPLE(1 VARCHAR NULL, 2 VARCHAR NULL) NULL ) ENGINE=FUSE
```
Comment (b41sh) [inline tests/sqllogictests/suites/base/05_ddl/05_0003_ddl_alter_table.test:133]: seems there's an extra space?

File: src/query/service/src/interpreters/interpreter_table_show_create.rs [hunk 2]
```diff
@@ -201,15 +196,10 @@ impl ShowCreateTableInterpreter {
...
-                    field.data_type().remove_recursive_nullable().sql_name(),
-                    nullable,
-                    default_expr,
-                    computed_expr,
-                    comment
-                );
+                let ident = display_ident(field.name(), quoted_ident_case_sensitive, sql_dialect);
+                let data_type = field.data_type().sql_name_explicit_null();
+                let column_str =
+                    format!("  {ident} {data_type}{default_expr}{computed_expr}{comment}");
```
Comment (TCeason) [inline src/query/service/src/interpreters/interpreter_table_show_create.rs:202]: Maybe extra space at start

File: tests/sqllogictests/suites/base/05_ddl/05_0003_ddl_alter_table.test (hunk 1 above)
Comment (forsaken628) [reply tests/sqllogictests/suites/base/05_ddl/05_0003_ddl_alter_table.test]: sqllogictests will ignore extra space. uh..  It's so weird.

**Review by TCeason — APPROVED** [review]

**Review by b41sh — APPROVED** [review]

**Pull request merged**